python/src/functions/generate_treasury_report.py

```diff
@@ -177,6 +178,10 @@ def process_event(payload: ProjectLambdaPayload, logger):
             with tempfile.NamedTemporaryFile() as file:
                 # Download projects from S3
                 created_at = file_info.createdAt
+
+                # TODO: Add a verification to ensure that the objectKey here is formatted correctly.
+                # Path must be: uploads/{organization_id}/{agency_id}/{reporting_period_id}/{upload_id}/{filename}.xlsm
```

Comment:
You could potentially borrow logic here from `matchRegex` in `processValidationJson`, which has the same key structure

Reply:
💯 - will address in a separate PR